scheduler/pkg/agent/repository/model_repository.go

```diff
@@ -138,6 +138,12 @@ func (r *V2ModelRepository) DownloadModelVersion(modelName string,
 		return nil, err
 	}
 
+	// Purge rclone path now we have loaded successfully
```

Comment:
there could be extra disk overhead incurred on main path because of the deletion and therefore maybe we could consider doing it async in a go routine perhaps?

Reply:
This would work most of the time but the name of the folder is not unique as it is a hash of model name and URI so under heavy load I suppose there could be a delete and an rclone sync at same time if the go routine to delete got delayed?

Reply:
Ok fair for this PR. We probably should still think about doing an async cleanup in a future PR.